src/odin/narrativebuilder.cc

```diff
@@ -3805,9 +3806,57 @@ std::string NarrativeBuilder::FormStreetNames(const StreetNames& street_names,
 void NarrativeBuilder::FormVerbalMultiCue(std::list<Maneuver>& maneuvers) {
   Maneuver* prev_maneuver = nullptr;
   for (auto& maneuver : maneuvers) {
-    if (prev_maneuver && IsVerbalMultiCuePossible(prev_maneuver, maneuver)) {
+    if (prev_maneuver && IsVerbalMultiCuePossible(*prev_maneuver, maneuver)) {
+      // Determine if imminent or distant verbal multi-cue
+      // if previous maneuver has an intersecting traversable outbound edge
+      // in the same direction as the maneuver
+      switch (maneuver.type()) {
+        case DirectionsLeg_Maneuver_Type_kSlightRight:
+        case DirectionsLeg_Maneuver_Type_kRight:
+        case DirectionsLeg_Maneuver_Type_kSharpRight:
+        case DirectionsLeg_Maneuver_Type_kUturnRight:
+        case DirectionsLeg_Maneuver_Type_kRampRight:
+        case DirectionsLeg_Maneuver_Type_kExitRight:
+        case DirectionsLeg_Maneuver_Type_kStayRight: {
+          if (prev_maneuver->has_right_traversable_outbound_intersecting_edge()) {
+            prev_maneuver->set_distant_verbal_multi_cue(true);
+          } else {
+            prev_maneuver->set_imminent_verbal_multi_cue(true);
+          }
+          break;
+        }
+        case DirectionsLeg_Maneuver_Type_kSlightLeft:
+        case DirectionsLeg_Maneuver_Type_kLeft:
+        case DirectionsLeg_Maneuver_Type_kSharpLeft:
+        case DirectionsLeg_Maneuver_Type_kUturnLeft:
+        case DirectionsLeg_Maneuver_Type_kRampLeft:
+        case DirectionsLeg_Maneuver_Type_kExitLeft:
+        case DirectionsLeg_Maneuver_Type_kStayLeft: {
+          if (prev_maneuver->has_left_traversable_outbound_intersecting_edge()) {
+            prev_maneuver->set_distant_verbal_multi_cue(true);
```

Comment:
Naming nit: I'm not sure if `distant` really captures the meaning. Whether a multicue is `imminent` or `distant` depends not on distance but on whether there is a traversable outbound intersecting edge in the same direction as the upcoming maneuver. The purpose is to disambiguate multiple possible turns by including a distance in the phrase, so what if we called it `disambiguated_multi_cue` or something along those lines?

Reply:
yes - I struggled with the naming for the same reason. Maybe a refactor to just be:
- imminent
- non_imminent